File: app/src/main/java/com/lostincontext/ruledetails/RuleDetailsComponent.java
package com.lostincontext.ruledetails;

import com.lostincontext.application.ApplicationComponent;
import com.lostincontext.commons.FragmentScope;

import dagger.Component;


@FragmentScope
@Component(modules = RuleDetailsPresenterModule.class, dependencies = ApplicationComponent.class)
public interface RuleDetailsComponent {

    void inject(RuleDetailsActivity activity);

}

File: app/src/main/java/com/lostincontext/ruledetails/RuleDetailsComponent.kt
package com.lostincontext.ruledetails

import com.lostincontext.application.ApplicationComponent
import com.lostincontext.commons.FragmentScope

import dagger.Component


@FragmentScope
@Component(modules = arrayOf(RuleDetailsPresenterModule::class),
           dependencies = arrayOf(ApplicationComponent::class))
interface RuleDetailsComponent {

    fun inject(activity: RuleDetailsActivity)

}